src/bin/pgcopydb/ld_stream.c

```diff
@@ -111,6 +111,15 @@ stream_init_specs(StreamSpecs *specs,
 				}
 			};
 
+			if (specs->slot.wal2jsonNumericAsString)
+			{
+				int index = options.count;
+
+				options.keywords[index] = "numeric-data-types-as-string";
+				options.values[index] = "true";
+				options.count += 1;
+			}
+
```

Comment:
Where is the memory allocated? I think I would prefer the following approach, if we're going to make the code as simple as possible:

``` c
			KeyVal options = {
				.count = specs->slot.wal2jsonNumericAsString ? 7 : 6,
				.keywords = {
					"format-version",
					"include-xids",
					"include-schemas",
					"include-transaction",
					"include-types",
					"filter-tables",
					"numeric-data-types-as-string"
				},
				.values = {
					"2",
					"true",
					"true",
					"true",
					"true",
					"pgcopydb.*",
					"true"
				}
			};
```

Reply:
Your suggestion is definitely cleaner, although a bit hacky. I applied it after adding some comments.

KeyVal struct contains an array of char pointers with 64 elements. We already have reserved space for 64 char* here. My choice to set one such pointer to a string literal address is a weird choice. As long as we do not try to update the string and cause undefined behavior, I thought that it was ok. 

https://github.com/dimitri/pgcopydb/blob/2c14034531f29c690830e4dd5e04f14771cc0f70/src/bin/pgcopydb/parsing_utils.h#L40-L59

Reply:
That comment seems outdated, we don't use MAXCONNINFO anymore (because libpq does not use it either), we use `strdup` instead.